### PR TITLE
[diffusion] refactor: remove hard-code of instanceof on PipelineConfig

### PR DESCRIPTION
**Pull request by mickqian**
## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.ai/developer_guide/contribution_guide.html#code-style-guidance).
- [ ] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

### CODE REVIEW

File: python/sglang/multimodal_gen/configs/pipeline_configs/flux.py [hunk 1]
```diff
@@ -464,6 +468,29 @@ def calculate_condition_image_size(
...
+        for latent in image_latents:
+            # latent: (1, 128, 32, 32)
+            packed = self.maybe_pack_latents(latent, None, None)  # (1, 1024, 128)
+            packed = packed.squeeze(0)  # (1024, 128) - remove batch dim
+            packed_latents.append(packed)
+
+        # Concatenate all reference tokens along sequence dimension
+        image_latents = torch.cat(packed_latents, dim=0)  # (N*1024, 128)
+        image_latents = image_latents.unsqueeze(0)  # (1, N*1024, 128)
+        image_latents = image_latents.repeat(batch_size, 1, 1)
```
[the remark this answers is not in the document]
Comment (mickqian) [reply python/sglang/multimodal_gen/configs/pipeline_configs/flux.py:487]: /gemini where is that `_prepare_image_ids` from?
